docker/elasticsearch-setup/create-indices.sh

```diff
@@ -14,22 +14,102 @@ else
+
+  #reindex the documents in source index to target index.
+  # One of the assumption here is that we only add properties to document when index-config is evolved.
+  # In case a property is deleted from document, it will still be reindexed in target index as default behaviour and
+  # it is not breaking the code. If still needs to be purged from target index, use "removed" property in POST data.
+  curl -XPOST "$ELASTICSEARCH_PROTOCOL://$ELASTICSEARCH_HOST_URL:$ELASTICSEARCH_PORT/_reindex?pretty" -H 'Content-Type: application/json' \
+    -d "{\"source\":{\"index\":\"$source_index\"},\"dest\":{\"index\":\"$target_index\"}}"
+
+  if check_reindex "$source_index" "$target_index"
+  then
+    #checking if source index is concrete index or alias
+    if [ $(curl -o /dev/null -s -w "%{http_code}" "$ELASTICSEARCH_PROTOCOL://$ELASTICSEARCH_HOST_URL:$ELASTICSEARCH_PORT/_alias/$source_index") -eq 404 ]
+    then
+      curl -XDELETE "$ELASTICSEARCH_PROTOCOL://$ELASTICSEARCH_HOST_URL:$ELASTICSEARCH_PORT/$source_index"
+    else
+      concrete_index_name=$(curl -XGET "$ELASTICSEARCH_PROTOCOL://$ELASTICSEARCH_HOST_URL:$ELASTICSEARCH_PORT/_alias/$source_index" | jq 'keys[]' | head -1 | tr -d \")
+      curl -XDELETE "$ELASTICSEARCH_PROTOCOL://$ELASTICSEARCH_HOST_URL:$ELASTICSEARCH_PORT/$concrete_index_name"
+    fi
+
+    curl -XPOST "$ELASTICSEARCH_PROTOCOL://$ELASTICSEARCH_HOST_URL:$ELASTICSEARCH_PORT/_aliases" -H 'Content-Type: application/json' \
+      -d "{\"actions\":[{\"remove\":{\"index\":\"*\",\"alias\":\"$source_index\"}},{\"add\":{\"index\":\"$target_index\",\"alias\":\"$source_index\"}}]}"
+
+    echo -e "\nReindexing to $target_index is complete"
+    return 0
+  else
+    echo -e "\nReindexing to $target_index is failure"
+    return 1
+  fi
+}
+
+function create_index() {
+  generate_index_file "index/$2" "index/$3" /tmp/data
+
+  #checking if index(or alias) exists
+  if [ $(curl -o /dev/null -s -w "%{http_code}" "$ELASTICSEARCH_PROTOCOL://$ELASTICSEARCH_HOST_URL:$ELASTICSEARCH_PORT/$1") -eq 404 ]
+  then
+    echo -e '\ncreating index' "$1"
+
+    curl -XPUT "$ELASTICSEARCH_PROTOCOL://$ELASTICSEARCH_HOST_URL:$ELASTICSEARCH_PORT/$1" -H 'Content-Type: application/json' --data @/tmp/data
+    return 0
+  else
+    echo -e '\ncomparing with existing version of index' "$1"
+
+    curl -XGET "$ELASTICSEARCH_PROTOCOL://$ELASTICSEARCH_HOST_URL:$ELASTICSEARCH_PORT/$1/_settings" | jq '.. | .settings? | select(. != null)' | \
+      jq '. | {"index":{"max_ngram_diff": .index.max_ngram_diff, "analysis": .index.analysis}}' > /tmp/existing_setting
+    curl -XGET "$ELASTICSEARCH_PROTOCOL://$ELASTICSEARCH_HOST_URL:$ELASTICSEARCH_PORT/$1/_mapping" | jq '.. | .mappings? | select(. != null)' > /tmp/existing_mapping
+
+    generate_index_file /tmp/existing_setting /tmp/existing_mapping /tmp/existing
+
+    if diff <(jq -S . /tmp/existing) <(jq -S . /tmp/data)
+    then
+      echo -e "\nexisting version of index $1 is latest"
```

Comment:
nit: no changes to index $1 mappings and settings

Reply:
done